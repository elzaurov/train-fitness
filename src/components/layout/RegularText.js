import React from 'react';
import PropTypes from 'prop-types';
import {Text, StyleSheet} from 'react-native';
import {Trans} from 'react-i18next';
import {Colors} from '../../constants';

const RegularText = ({style, children, i18nKey, ...props}) => (
  <Text style={[styles.text, style]} {...props}>
    {i18nKey ? <Trans i18nKey={i18nKey}>{children}</Trans> : children}
  </Text>
);

RegularText.propTypes = {
  style: PropTypes.any,
  children: PropTypes.any,
  i18nKey: PropTypes.string,
};

RegularText.defaultProps = {
  style: {},
  children: undefined,
  i18nKey: undefined,
};

export default RegularText;

const styles = StyleSheet.create({
  text: {
    color: Colors.white,
    fontFamily: 'TitilliumWeb-Regular',
  },
});
