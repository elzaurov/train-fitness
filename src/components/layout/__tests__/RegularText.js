import React from 'react';
import renderer from 'react-test-renderer';
import RegularText from '../RegularText';

test('Should render "test" Text', () => {
  const tree = renderer.create(<RegularText>test</RegularText>).toJSON();
  expect(tree).toMatchSnapshot();
});
