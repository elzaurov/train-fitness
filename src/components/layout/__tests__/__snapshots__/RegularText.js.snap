// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render "test" Text 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#fff",
        "fontFamily": "TitilliumWeb-Regular",
      },
      Object {},
    ]
  }
>
  test
</Text>
`;
